pkg/util/util.go

```diff
@@ -1144,6 +1150,49 @@ func ValidateFile(filePath string) error {
 	return nil
 }
 
+// GetGitUrlComponentsFromRaw converts a raw GitHub file link to a map of the url components
+func GetGitUrlComponentsFromRaw(rawGitURL string) (map[string]string, error) {
+	err := ValidateURL(rawGitURL)
+	if err != nil {
+		return nil, err
+	}
+	if !strings.Contains(rawGitURL, "raw.githubusercontent.com") {
```

Comment:
you have checked this before calling the function, no need to check it again

Reply:
removed